### DISCUSSION TITLE
Change Bezier Spline in runtime

### DISCUSSION THREAD

**Discussion by mimpedro** [yasirkula/UnityBezierSolution#19] [Q&A] [answered]
Firstly, thank you so much for this package! It is by far one of the best and more friendly bezier tools I've seen in Unity.

I have a bezierWalker that I want to jump from one bezierPath to a different one depending on certain conditions, but when trying to change the spline it is running on, I noticed the bezierWalker.Spline value is read-only.

What is the recommended way of achieving this? Having multiple walkers on the same object (one for each spline) and activating/deactivating them? Or is there a way to use a single walker and change its spline in runtime?

**Comment by yasirkula** [comment, marked as answer]
Hi! You can change the **spline** field (not the **Spline** property) to switch splines. If your variable's type is BezierWalker, then you should change it to either BezierWalkerWithSpeed or BezierWalkerWithTime because the field exists on those classes only.

**Reply by mimpedro** [reply]
Oh, I see! I was using the base class in my code, this is why I wasn't seeing it!

Thanks a lot :)
